### ISSUE TITLE
Session Credentials do not work

### ISSUE THREAD

**Issue by rjezek** [aws/aws-sdk-net#785]
I have valid temporary credentials from STS in my credentials file and everything from the CLI works just fine.  When trying to run code from the SDK it says my token is invalid.

I would like to see the SDK support temporary credentials like the CLI does.

**Comment by rjezek** [comment]
Just to add some clarity... when you use temporary credentials your credentials file has this format:

[default]
aws_access_key_id = xxx
aws_secret_access_key =  xxx
aws_security_token = xxx

**Comment by spati2** [comment]
What is the AWSSDK.Core version that you are using?

**Comment by rjezek** [comment]
AWSSDK.Core 3.3.17

I can authenticate via the SDK using SessionAWSCredentials and setting up my credentials manually.  I am specifically asking about the SDK supporting this via the credentials file like the CLI does.

**Comment by Bluestreak2k5** [comment]
@rjezek Are you using a proxy in your Config's? I ran into this same exact issue when I was using Visual Studio with AWS toolkit.

For Azure I had to set this in the devenv..exe.config file for Visual Studio:
	  <defaultProxy useDefaultCredentials="true" enabled="true">
		<proxy bypassonlocal="true" proxyaddress="http://webaddress" />
	  </defaultProxy>
However, once I tried using AWS toolkit the proxy setting didn't seem to work. Removing the proxy settings seemed to make the credentials valid for whatever reason.
